### ISSUE TITLE
Issues - Win 10

### ISSUE THREAD

**Issue by dereuromark** [spryker/devvm#8]
I tested it in win10 and everything seems to work,
I can also run commands like `vendor/bin/console ...`

But the frontends yves and zed only output

```
File not found.
```

and the setup-vm call outputs lots of npm errors when running

```
m ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "spy-zed"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4
npm ERR! code ELIFECYCLE
npm ERR! yves_zed@0.3.0 spy-zed: `gulp --gulpfile gulp/zed.js --cwd .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the yves_zed@0.3.0 spy-zed script 'gulp --gulpfile gulp/zed.js --cwd .'.
npm ERR! This is most likely a problem with the yves_zed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp --gulpfile gulp/zed.js --cwd .
npm ERR! You can get their info via:
npm ERR!     npm owner ls yves_zed
npm ERR! There is likely additional logging output above.
```

**Comment by marek-obuchowicz** [comment]
@dereuromark does the mentioned command, `gulp --gulpfile gulp/zed.js --cwd .`, give any helpful output?

**Comment by dereuromark** [comment]
I get
```
/data/shop/development/current/node_modules/gulp-sass/node_modules/node-sass/lib/extensions.js:158
    throw new Error([
    ^

Error: The `libsass` binding was not found in /data/shop/development/current/node_modules/gulp-sass/node_modules/node-sass/vendor/linux-x64-47/binding.node
This usually happens because your node version has changed.
Run `npm rebuild node-sass` to build the binding for your current node version.
    at Object.sass.getBinaryPath (/data/shop/development/current/node_modules/gulp-sass/node_modules/node-sass/lib/extensions.js:158:11)
    at Object.<anonymous> (/data/shop/development/current/node_modules/gulp-sass/node_modules/node-sass/lib/index.js:16:36)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/data/shop/development/current/node_modules/gulp-sass/index.js:186:21)
    at Module._compile (module.js:413:34)
```
Same issue with the 43 current latest VM by the way.
Everything else seems to run fine, only the final browser output doesnt.

I also saw this

    Last login: Mon Mar  7 11:29:44 2016 from 10.0.2.2

And therefore added

    10.0.2.2 spryker.dev zed.de.spryker.dev zed.com.spryker.dev www.com.spryker.dev com.spryker.dev static.com.spryker.dev www.de.spryker.dev de.spryker.dev static.de.spryker.dev kibana.spryker.dev

to the hosts file just in case, because so far there was only 10.10.0.33 configured there.
But also didnt help, of course. And from the looks it is only the asset stuff not working, the routing to the Spryker demoshop seems to be fine, otherwise the "File not found." would not show up, right?

**Comment by dereuromark** [comment]
Trying to do the proposed `npm rebuild node-sass` errors with

```
vagrant@spryker-vagrant ?  current git:(master)  npm rebuild node-sass
npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "rebuild" "node-sass"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! path ../node-sass/bin/node-sass
npm ERR! code EPROTO
npm ERR! errno -71
npm ERR! syscall symlink

npm ERR! EPROTO: protocol error, symlink '../node-sass/bin/node-sass' -> '/data/shop/development/current/node_modules/gulp-sass/node_modules/.bin/node-sass'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /data/shop/development/current/npm-debug.log
```

**Comment by marek-obuchowicz** [comment]
Please have a look on https://github.com/npm/npm/issues/9901. `npm` requires additional options to be able to run on windows - `--no-bin-links`. The VM provides just basic node.js setup, it doesn't handle how we call npm. Can you please test if `npm rebuild node-sass` works with this flag and, if possible, adjust `setup-frontend.sh` file (https://github.com/spryker/demoshop/blob/develop/setup-frontend.sh)? (it's not part of dev vm, but spryker code)

**Comment by dereuromark** [comment]
It ran quite far this time, but then
```
npm WARN bl@0.7.0 requires a peer of stream-browserify@* but none was installed.
npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! path ../browserslist/cli.js
npm ERR! code EPROTO
npm ERR! errno -71
npm ERR! syscall symlink

npm ERR! EPROTO: protocol error, symlink '../browserslist/cli.js' -> '/data/shop/development/current/node_modules/.bin/browserslist'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /data/shop/development/current/npm-debug.log

npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "spy-setup"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE
npm ERR! yves_zed@0.3.0 spy-setup: `npm install && npm run spy-yves && npm run spy-zed`
npm ERR! Exit status 185
npm ERR!
npm ERR! Failed at the yves_zed@0.3.0 spy-setup script 'npm install && npm run spy-yves && npm run spy-zed'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the yves_zed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm install && npm run spy-yves && npm run spy-zed
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs yves_zed
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls yves_zed
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /data/shop/development/current/npm-debug.log
```

**Comment by marek-obuchowicz** [comment]
https://github.com/spryker/demoshop/blob/master/package.json#L15 - there's a call to `npm install`, it requires flag `--no-bin-links` as well

**Comment by dereuromark** [comment]
spy-zed works, but the other ones not quite yet:

```
npm WARN bl@0.7.0 requires a peer of stream-browserify@* but none was installed.
npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--no-bin-links"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! path /data/shop/development/current/node_modules/gulp-concat/node_modules/gulp-util/node_modules/dateformat/node_modules/meow/node_modules/normalize-package-data/node_modules/validate-npm-package-license/node_modules/spdx-expression-parse/node_modules/spdx-license-ids
npm ERR! code EEXIST
npm ERR! errno -17
npm ERR! syscall rename

npm ERR! EEXIST: file already exists, rename '/data/shop/development/current/node_modules/gulp-concat/node_modules/gulp-util/node_modules/dateformat/node_modules/meow/node_modules/normalize-package-data/node_modules/validate-npm-package-license/node_modules/spdx-expression-parse/node_modules/spdx-license-ids' -> '/data/shop/development/current/node_modules/gulp-concat/node_modules/gulp-util/node_modules/dateformat/node_modules/meow/node_modules/normalize-package-data/node_modules/validate-npm-package-license/node_modules/spdx-expression-parse/node_modules/.spdx-license-ids.DELETE'
npm ERR! File exists: /data/shop/development/current/node_modules/gulp-concat/node_modules/gulp-util/node_modules/dateformat/node_modules/meow/node_modules/normalize-package-data/node_modules/validate-npm-package-license/node_modules/spdx-expression-parse/node_modules/spdx-license-ids
npm ERR! Move it away, and try again.

npm ERR! Please include the following file with any support request:
npm ERR!     /data/shop/development/current/npm-debug.log

npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "spy-setup"
npm ERR! node v5.6.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE
npm ERR! yves_zed@0.3.0 spy-setup: `npm install --no-bin-links && npm run spy-yves && npm run spy-zed`
npm ERR! Exit status 239
npm ERR!
npm ERR! Failed at the yves_zed@0.3.0 spy-setup script 'npm install --no-bin-links && npm run spy-yves && npm run spy-zed'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the yves_zed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm install --no-bin-links && npm run spy-yves && npm run spy-zed
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs yves_zed
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls yves_zed
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /data/shop/development/current/npm-debug.log
```

shouldnt they properly overwrite automatically?

**Comment by PierreAtRose** [comment]
Hey there,

the same problems which seem to depend on the missing symlinks really. The `--no-bin-links` parameter helps to get further but that's not enough.

404 is caused by the missing public folder in static.

To fix the NPM errors  i tried the follwing `npm --no-bin-links  install node-sass` first followed by `gulp --gulpfile config/Yves/gulpfile.js --cwd`. After that `npm --no-bin-links install && npm run spy-yves && npm run spy-zed "all"` runs without errors and the public folder exists but 'static' is missing in the path. Editing the webroot in the nginx config helps to see something but it seems to be not enough :)

Yves is reachable with some placeholders. Zed is reachable tool but the request on css and js files get a 500 status.

A clean solution would be nice.

Thx and greets

Pierre

**Comment by dereuromark** [comment]
So I got it to work by switching from dev-master to dev-develop.
The recent spy changes there seem to now display at least the ZED form, and the Yves frontpage.
Only the assets are not there yet (white HTML basiscs only).
So the assets error still. I get a few errors when running spy-setup etc.

**Comment by marek-obuchowicz** [comment]
Please test with latest release: https://github.com/spryker/devvm/releases/download/ci-70/spryker-devvm.box - no more npm problems are there, but we still need a proper testing for whole process managed by antilope

**Comment by hhebbo** [comment]
fixed

**Comment by v-v-volkov** [comment]
The same issue occurs again on WIndows 10. Can someone re-check that --no-bin-links is still used?

**Comment by marek-obuchowicz** [comment]
@abelspry do you have any feedback about this?

**Comment by v-v-volkov** [comment]
TL;DR we need to make sure that we use --no-bin-links everywhere in npm code, otherwise the demo sho won't be compatible with Windows 

We might also use this solution as alternative: http://blog.rudylee.com/2014/10/27/symbolic-links-with-vagrant-windows/

But we need someone to test it. My laptop takes ages to build the devvm.

@abelspry

**Comment by marek-obuchowicz** [comment]
@v-v-volkov question from another scope, what does it mean "takes ages to build the devvm", can you be more specific like: "it takes 30 minutes to execute `vendor/bin/install`? Any information about performance experience is helpful!

**Comment by v-v-volkov** [comment]
@marek-obuchowicz `vendor/bin/install` takes around 1.5-2 hours (and pay attention that Yves and Zed frontends are not fully installed due to the npm issues). Specs: 

Intel Core i5-7200U@2.50 GHz
8 GB RAM
256GB OPAL2 SSD
Windows 10 Pro all updates installed

**Comment by v-v-volkov** [comment]
No, the Dev VM release candidate solves the issue.
